include/experimental/__p1673_bits/blas3_matrix_product.hpp

```diff
@@ -673,7 +673,7 @@ void triangular_matrix_right_product(
       const ptrdiff_t k_upper = explicitDiagonal ? j : j - size_type(1);
```

Comment:
The variable `j` has type `size_type` (`std::size_t`, an unsigned integer), and `j - size_type(1)` has type `size_type` (please see [this godbolt example](https://godbolt.org/z/fe6YEfPb3)), so `k_upper` will be the biggest possible 64-bit positive integer.  The resulting `k <= k_upper` comparison will always evaluate to `true`.

The work-around that comes to mind would be to have a special case for `explicitDiagonal` being `true`.  That way, one could rewrite the loop bound as `k < j`.  Here is a possible reimplementation of lines 676-681:
```
for (ptrdiff_t k = 0; k < j; ++k) {
  C(i,j) += B(i,k) * A(k,j);
}
if constexpr (explicitDiagonal) {
  C(i,j) += B(i,j) /* times 1 */;
} else {
  C(i,j) += B(i,j) * A(j,j);
}
```

Reply:
Is changing size_type(1) to ptrdiff_t(1) acceptable? If not, I can change it.

Reply:
@amklinv-nnl Please see suggested changes below -- thanks!